### PR TITLE
update to 1.21.0

### PR DESCRIPTION
**Pull request by cbouss**
Jira: https://anaconda.atlassian.net/browse/DSNC-4930
Repo: https://github.com/pygobject/pycairo/
https://github.com/pygobject/pycairo/blob/v1.21.0/meson.build
https://github.com/pygobject/pycairo/blob/v1.21.0/meson_options.txt

Changes:
- Update version to 1.21.0
- Update meta data
- Change build to meson
- Fix windows install due to our old meson verison
- Add pkgconfig tests.

### CODE REVIEW

**Comment by cbouss** [comment]
TODO: remove conda_build_config.yaml before merging.

**Review by ScottBailey — APPROVED** [review]
In bld.bat we call ninja directly to run the build; however, in build.sh we call meson (which must call ninja). Would it be better if these patterns were the same?

Also, I notice we call `meson test` in build.sh but not bld.bat. Is this intentional?

Other than these questions, it looks fine to me.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,32 +10,42 @@ source:
...
   sha256: {{ sha256 }}
   patches:
-    - windows.patch  # [win]
+    - typing-test-workaround.patch
 
 build:
   number: 0
   detect_binary_files_with_prefix: true
-  skip: true  # [py < 35]
+  skip: true  # [py < 37]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: `py < 37` should be without whitespaces `py<37`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,32 +10,42 @@ source:
...
-    - m2w64-pkg-config  # [win]
-    - cairo
-    - cython
-    - pip
-    - pthread-stubs
+    - cairo >=1.15.10
+    - flake8  # for testing during the build
+    - glib
+    - pytest  # for testing during the build
     - python
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:43]: It requires `setuptools` and `wheel` in `host`, see https://github.com/pygobject/pycairo/blob/7f7869f4139dbcc84a6f688a321516cb25cea1b0/pyproject.toml#L21

**Review by tobijk — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: thank you - addressed by new commit.

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml:43]: thank you - addressed by new commit.

**Comment by cbouss** [comment]
> In bld.bat we call ninja directly to run the build; however, in build.sh we call meson (which must call ninja). Would it be better if these patterns were the same?
> 
> Also, I notice we call `meson test` in build.sh but not bld.bat. Is this intentional?
> 
> Other than these questions, it looks fine to me.

Thank you. I took the win build structure from another feedstock. I have aligned bld.bat and build.sh via a new commit.

**Comment by cbouss** [comment]
Now linux-ppc64le stopped building on prefect while it still builds on concourse. From the logs, something is pulling libxcb implicitly. libxcb is used instead of the cdt version. However libxcb pc file points to xau and the build cannot find the xau .pc file.

Edit: same result on concourse after I pull the latest cbc from aggregate.
Edit: this is due to the last compiler update.

**Pull request merged**